src/c4/test/tstifpstream.cc

```diff
@@ -0,0 +1,105 @@
+//----------------------------------*-C++-*-----------------------------------//
+/*!
+ * \file   c4/test/tstifpstream.cc
+ * \author Mathew Cleveland
+ * \date   Feb. 2020
+ * \brief  Tests ifpstream class for paralle read
+ * \note   Copyright (C) 2016-2020 Triad National Security, LLC.
+ *         All rights reserved. */
+//----------------------------------------------------------------------------//
+
+#include "c4/ParallelUnitTest.hh"
+#include "c4/ifpstream.hh"
+#include "c4/ofpstream.hh"
+#include "ds++/Release.hh"
+#include <sstream>
+
+using namespace std;
+using namespace rtt_dsxx;
+using namespace rtt_c4;
+
+//----------------------------------------------------------------------------//
+// HELPER FUNCTIONS
+//----------------------------------------------------------------------------//
+
+void write_stream(std::string filename, const std::ios_base::openmode mode) {
+
+  unsigned const pid = rtt_c4::node();
+
```

Comment:
For unsigned node values you might consider using `rtt_c4::rank()`.

Reply:
Fixed.